pymongo/database.py

```diff
@@ -1079,19 +1168,24 @@ def list_collection_names(
 
         return [result["name"] for result in self.list_collections(session=session, **kwargs)]
 
-    def _drop_helper(self, name, session=None, comment=None):
+    def _drop_helper(
+        self, name: str, session: Optional[ClientSession] = None, comment: Optional[Any] = None
+    ) -> Dict[str, Any]:
         command = SON([("drop", name)])
         if comment is not None:
             command["comment"] = comment
 
         with self.__client._socket_for_writes(session) as sock_info:
-            return self._command(
-                sock_info,
-                command,
-                allowable_errors=["ns not found", 26],
-                write_concern=self._write_concern_for(session),
-                parse_write_concern_error=True,
-                session=session,
+            return cast(
+                Dict[str, Any],
```

Comment:
Cast no longer needed if we default _command to return Dict (the concrete type).